src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs

```diff
@@ -64,6 +66,39 @@ internal struct NameDeclarationInfo
                 return default;
             }
 
+            private static bool IsTupleElementDeclaration(
+                SyntaxToken token, SemanticModel semanticModel, int position,
+                CancellationToken cancellationToken, out NameDeclarationInfo result)
+            {
+                result = IsFollowingTypeOrComma<TupleElementSyntax>(
+                    token,
+                    semanticModel,
+                    tupleElement => tupleElement.Type,
+                    _ => default(SyntaxTokenList),
+                    _ => ImmutableArray.Create(SymbolKind.Local), cancellationToken);
+
+                return result.Type != null;
+            }
+
+            private static bool IsTupleExpressionDeclaration(
```

Comment:
It would be good to add a comment to clarify that this is for handling incomplete code that parsed like a tuple literal.
But the purpose is not to help typing names in legit tuple literals (that's not possible since they don't have any TypeSyntax).

Reply:
Done cf1a9739.